skydoc/macro_extractor.py

```diff
@@ -87,7 +89,7 @@ def _add_macro_doc(self, stmt):
 
     for i in range(len(stmt.args.args)):
       attr = rule.attribute.add()
-      attr_name = stmt.args.args[i].id
+      attr_name = stmt.args.args[i].id if hasattr(stmt.args.args[i], 'id') else stmt.args.args[i].arg
```

Comment:
@philwo Do you know why we need this change? I was struggling with a `'arg' Object has no attribute 'id'` error.

Reply:
Yes, the ast library unfortunately slightly changed between Py2 and Py3:

In Python 2, the "arguments" type looks like this:

```
	arguments = (expr* args, identifier? vararg, 
		     identifier? kwarg, expr* defaults)

	expr = [...]
	     | Name(identifier id, expr_context ctx)
```

But in Python 3 it's like this:

```
    arguments = (arg* args, arg? vararg, arg* kwonlyargs, expr* kw_defaults,
                 arg? kwarg, expr* defaults)

    arg = (identifier arg, expr? annotation)
           attributes (int lineno, int col_offset
```

Because Python 2 uses an "expr" to reference arguments, we have to go through the "id" field to get the name, but in Python 3 we have to go through the new custom "arg" type and get the name from its "arg" field.

Reply:
Ah, thanks for the explanation!